### PR TITLE
feat: Handle request for user verified public key for evm

### PR DESCRIPTION
**Pull request by natto1784**
Changes included:
- Added handling for getting user verified public key
- Made code uniform with the near app in https://github.com/Cypherock/x1_wallet_firmware/pull/372

Depends on https://github.com/Cypherock/cypherock-common/pull/33

https://app.clickup.com/t/9002019994/PRF-4896

### CODE REVIEW

File: apps/evm_family/evm_pub_key.c [hunk 1]
```diff
@@ -345,28 +402,42 @@ void evm_get_pub_keys(evm_query_t *query) {
                              &seed[0])) {
     memzero(seed, sizeof(seed));
     // TODO: Handle error case reporting to host
+    memzero(seed, sizeof(seed));
```
Comment (kartikCypherock) [inline apps/evm_family/evm_pub_key.c:405]: Same statement is repeated.. pls check again

File: apps/evm_family/evm_pub_key.c [hunk 2]
```diff
@@ -345,28 +402,42 @@ void evm_get_pub_keys(evm_query_t *query) {
...
-  bool status = get_public_keys(init_req->derivation_paths,
-                                init_req->derivation_paths_count,
-                                seed,
-                                public_keys);
+  if (!fill_public_keys(init_req->derivation_paths,
+                        seed,
+                        public_keys,
+                        init_req->derivation_paths_count)) {
+    // Clear seed as soon as it is not needed
+    memzero(seed, sizeof(seed));
```
Comment (kartikCypherock) [inline apps/evm_family/evm_pub_key.c:416]: Seed will not be cleared in happy flow, please re-check

Comment (natto1784) [reply apps/evm_family/evm_pub_key.c]: fixed

File: apps/evm_family/evm_pub_key.c [hunk 3]
```diff
@@ -343,30 +400,44 @@ void evm_get_pub_keys(evm_query_t *query) {
...
     evm_send_error(ERROR_COMMON_ERROR_UNKNOWN_ERROR_TAG, 1);
+    return;
   }
 
-  if (status) {
-    delay_scr_init(ui_text_check_cysync_app, DELAY_TIME);
+  if (EVM_QUERY_GET_USER_VERIFIED_PUBLIC_KEY_TAG == which_request) {
+    char address[100] = "";
+    byte_array_to_hex_string(
+        public_keys[0], sizeof(public_keys[0]), address, sizeof(address));
```
Comment (ujjwal-cyph) [inline apps/evm_family/evm_pub_key.c:427]: need to encode the ethereum address. ref: `hdnode_get_ethereum_pubkeyhash` and `ethereum_address_checksum`

**Review by ujjwal-cyph — CHANGES_REQUESTED** [review]
resolve submodule conflict

File: apps/evm_family/evm_pub_key.c [hunk 4]
```diff
@@ -343,30 +400,44 @@ void evm_get_pub_keys(evm_query_t *query) {
...
+    if (!core_scroll_page(ui_text_receive_on, address, evm_send_error)) {
+      return;
+    }
+    core_status_set_flow_status(EVM_GET_PUBLIC_KEYS_STATUS_VERIFY);
```
Comment (kartikCypherock) [inline apps/evm_family/evm_pub_key.c:431]: You will also need to rebase with base branch as this function is to be replaced with `set_app_flow_status`

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4896 Ethereum get public key function](https://app.clickup.com/t/9002019994/PRF-4896)

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**